app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/health/AppTPCPUMonitor.kt

```diff
@@ -0,0 +1,115 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.mobile.android.vpn.health
+
+import android.content.Context
+import android.os.SystemClock
+import android.system.Os
+import android.system.OsConstants
+import androidx.work.ExistingPeriodicWorkPolicy
+import androidx.work.PeriodicWorkRequestBuilder
+import androidx.work.WorkInfo.State.CANCELLED
+import androidx.work.WorkManager
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.mobile.android.vpn.pixels.DeviceShieldPixels
+import com.squareup.anvil.annotations.ContributesBinding
+import dagger.SingleInstanceIn
+import timber.log.Timber
+import java.io.BufferedReader
+import java.io.File
+import java.io.FileReader
+import java.io.IOException
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+import kotlin.math.roundToInt
+import kotlin.time.Duration.Companion.seconds
+
+@SingleInstanceIn(AppScope::class)
+@ContributesBinding(AppScope::class)
+class AppTPCPUMonitor @Inject constructor(
+    private val workManager: WorkManager,
+): CPUMonitor {
+
+    companion object {
+        private val CLOCK_SPEED_HZ = Os.sysconf(OsConstants._SC_CLK_TCK)
+        private val NUM_CORES = Os.sysconf(OsConstants._SC_NPROCESSORS_CONF)
+
+        private val WHITE_SPACE = "\\s".toRegex()
+
+        // Indices in /proc/[pid]/stat (https://linux.die.net/man/5/proc)
+        private val UTIME_IDX = 13
+        private val STIME_IDX = 14
+        private val STARTTIME_IDX = 21
+        private val PROC_SIZE = 44
+    }
+
+    override fun startMonitoring() {
+        Timber.v("AppTPCPU - startMonitoring")
+        val work = PeriodicWorkRequestBuilder<CPUMonitorWorker>(15, TimeUnit.MINUTES)
+            .build()
+
+        workManager.enqueueUniquePeriodicWork("cpuMon", ExistingPeriodicWorkPolicy.KEEP, work)
+    }
+
+    override fun stopMonitoring() {
+        Timber.v("AppTPCPU - stopMonitoring")
+        workManager.cancelUniqueWork("cpuMon")
+    }
+
+    override fun isMonitoringStarted(): Boolean {
+        val listenableFuture = workManager.getWorkInfosForUniqueWork("cpuMon")
+        val workerList = listenableFuture.get()
+
+        return workerList != null && workerList.size > 0 && workerList[0].state != CANCELLED
+    }
+
+    class CPUMonitorWorker(context: Context,
+        workerParams: WorkerParameters
+    ) : Worker(context, workerParams) {
+        @Inject
+        lateinit var deviceShieldPixels: DeviceShieldPixels
```

Comment:
This dependency will not be injected by our DI and will be null.
For Android workers to be part of DI you'll need to annotate them with `ContributesWorker`. That annotation will code-gen the necessary glue code so that then DI dependencies are properly injected.